src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -295,7 +301,62 @@ public static Complex Exp(Complex value)
         [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Sqrt", Justification = "Sqrt is the name of a mathematical function.")]
         public static Complex Sqrt(Complex value)
         {
-            return FromPolarCoordinates(Math.Sqrt(value.Magnitude), value.Phase / 2.0);
+
+            if (value._imaginary == 0.0)
+            {
+                // Handle the trivial case quickly.
+                if (value._real < 0.0)
+                {
+                    return new Complex(0.0, Math.Sqrt(-value._real));
+                }
+                else
+                {
+                    return new Complex(Math.Sqrt(value._real), 0.0);
+                }
+            }
+            else
```

Comment:
In my mind, early return as much as possible makes the code easier to read, especially if there are nested cases and later the code in the block gets longer so you have to scroll up and figure out what case you are in. Yes a nit only.